### ISSUE TITLE
Bug with Tauri

### ISSUE THREAD

**Issue by reyamir** [brillout/vite-plugin-ssr#808]
### Description

build process run successful without any erros, but when I run app, I got this error

### Error Message + Error Stack

```shell
[Error] Error: [vite-plugin-ssr@0.4.115][Bug] You stumbled upon a bug in vite-plugin-ssr's source code. Go to https://github.com/brillout/vite-plugin-ssr/issues/new and copy-paste this error. (The error's stack trace is usually enough to fix the problem). A maintainer will fix the bug (usually under 24 hours). Don't hesitate to reach out as it makes vite-plugin-ssr more robust. Debug info (this is for the vite-plugin-ssr maintainers; you can ignore this): `{"url":"tauri://localhost"}`

	r (dd5696a8.js:2:2346)
	Fe (dd5696a8.js:2:4988)
	Wt (dd5696a8.js:2:7137)
	Module Code (38f660e8.js:2:3820)
```

**Comment by brillout** [comment]
I see. Let me have a closer look.

**Comment by brillout** [comment]
I cannot build:

```
error during build:
SyntaxError: Unexpected token (1:6431) in /workspace/lume/node_modules/.pnpm/unique-names-generator@4.7.1/node_modules/unique-names-generator/dist/index.m.js
```

I'm trying on https://github.com/luminous-devs/lume/tree/ditch-nextjs.

Is the latest code pushed?

**Comment by reyamir** [comment]
I've pushed new commit, you can run build again

add `--debug` flag to active console

```
pnpm tauri build --debug
```

**Comment by brillout** [comment]
Still no luck:

```
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:warning=`PKG_CONFIG_ALLOW_SYSTEM_CFLAGS="1" "pkg-config" "--libs" "--cflags" "javascriptcoregtk-4.0" "javascriptcoregtk-4.0 >= 2.24"` did not exit successfully: exit status: 1
  error: could not find system library 'javascriptcoregtk-4.0' required by the 'javascriptcore-rs-sys' crate

  --- stderr
  Package javascriptcoregtk-4.0 was not found in the pkg-config search path.
  Perhaps you should add the directory containing `javascriptcoregtk-4.0.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'javascriptcoregtk-4.0' found
  Package javascriptcoregtk-4.0 was not found in the pkg-config search path.
  Perhaps you should add the directory containing `javascriptcoregtk-4.0.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'javascriptcoregtk-4.0' found

       Error failed to build app: failed to build app
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by brillout** [comment]
The steps I took:
 1. https://gitpod.io#https://github.com/luminous-devs/lume/tree/ditch-nextjs
 2. `pnpm install` (Gitpod does it automatically)
 3. `pnpm tauri build --debug`

**Comment by reyamir** [comment]
oh, I'm not sure gitpod support build tauri app, because it require some native library, you should try on computer

**Comment by brillout** [comment]
With Gitpod, the code doesn't run in the browser but in a virtual machine. So AFAICT it should work on Gitpod.

>   Package javascriptcoregtk-4.0 was not found in the pkg-config search path.

It doesn't seem to be Gitpod related.

**Comment by reyamir** [comment]
I've pushed new commit which add some addition files for build on gitpod, can you try again

reference: [2022-tauri-gitpod-example](https://github.com/schickling-test/2022-tauri-gitpod-example)

**Comment by brillout** [comment]
It's now building but I get this in the browser:

```
TypeError: window.__TAURI_IPC__ is not a function
```

**Comment by brillout** [comment]
When running the command https://github.com/luminous-devs/lume/blob/451fc152521a0e2cac02545984b56243e3ef891b/.gitpod.yml#L6.

**Comment by reyamir** [comment]
I've managed run `tauri` with `gitpod` in this workspace https://luminousdevs-lume-rnri9cgz29x.ws-us95.gitpod.io

the process `pnpm tauri build` run successfully after I install some missing packages, but after that `tauri` will auto trigger installer for the build package (.AppImage on Linux and .dmg on Mac), but gitpod don't support install desktop package, so this process will failed after success build

the only way to re proceduce my issue is running build on real machine

all commands that I run:

```
sudo apt update
```

```
sudo apt install build-essential libwebkit2gtk-4.0-dev libssl-dev libgtk-3-dev libjavascriptcoregtk-4.0-dev librsvg2-dev at-spi2-core libsoup2.4-dev libayatana-appindicator3-dev
```

```
pnpm tauri build --debug
```

<img width="1203" alt="Screenshot 2023-04-23 at 21 57 16" src="https://user-images.githubusercontent.com/123083837/233847192-671f3f4a-421b-4ac8-8ca6-59cf4b99ac2a.png">

**Comment by brillout** [comment]
I see. Let me try locally.

**Comment by brillout** [comment]
No luck locally either:

```
    Finished dev [unoptimized + debuginfo] target(s) in 18m 40s
    Bundling lume_0.2.5_amd64.deb (/home/rom/tmp/lume/src-tauri/target/debug/bundle/deb/lume_0.2.5_amd64.deb)
    Bundling lume_0.2.5_amd64.AppImage (/home/rom/tmp/lume/src-tauri/target/debug/bundle/appimage/lume_0.2.5_amd64.AppImage)
       Error failed to bundle project: error running appimage.sh: error running appimage.sh: `failed to run /home/rom/tmp/lume/src-tauri/target/debug/bundle/appimage/build_appimage.sh`
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by brillout** [comment]
> [Error] Error: [vite-plugin-ssr@0.4.115][Bug] You stumbled upon a bug in vite-plugin-ssr's source code. Go to https://github.com/brillout/vite-plugin-ssr/issues/new and copy-paste this error. (The error's stack trace is usually enough to fix the problem). A maintainer will fix the bug (usually under 24 hours). Don't hesitate to reach out as it makes vite-plugin-ssr more robust. Debug info (this is for the vite-plugin-ssr maintainers; you can ignore this): `{"url":"tauri://localhost"}`

This means that an [`assert()`](https://github.com/brillout/vite-plugin-ssr/blob/fefed3266c9432f45a8950e5ec4e32d735cf9777/vite-plugin-ssr/utils/assert.ts#L24) call is failing.

> r (dd5696a8.js:2:2346)

What's the content of that line? Some call in the strack trace will be an `assert()` call; I need the content of the `assert()` line and the function name calling `assert()`.

It could be an assert in [parseUrl.ts](https://github.com/brillout/vite-plugin-ssr/blob/fefed3266c9432f45a8950e5ec4e32d735cf9777/vite-plugin-ssr/utils/parseUrl.ts).

The fix should be relatively easy then: basically treating URLs such as `tauri://localhost` like `https://localhost` (currently VPS assumes that URLs start with either `http://`, `https://`, or `/`). In case you'd be up to try to write a vite-plugin-ssr patch I'm happy to assist (although I don't know if other errors will pop up after we fix this one).

The best would still be that I fix it myself. Beyond the `failed to run build_appimage.sh` error, is there a way for me to try Tauri in some kind of dev mode against a linked `node_modules/vite-plugin-ssr/`? Building the whole Tauri app takes quite some time and doing that everytime I make a change to vite-plugin-ssr's source would be quite a slow process.

**Comment by brillout** [comment]
I managed to make the Tauri build work.

**Comment by brillout** [comment]
Alright, it's done.

![Screenshot 2023-04-23 23 47 37](https://user-images.githubusercontent.com/1005638/233869697-1b8bf484-75c9-4d3e-9a7f-e62d66a2ad2b.png)

The fixes are released in `0.4.116`.

Lume is open source so I guess you don't have much resources, but if you (eventually) do: https://github.com/sponsors/brillout.

Looking forward to the Tauri + VPS example. Unfortunately there's likely no easy way to add Tauri to VPS's CI, so I'd suggest we keep the example at `https://github.com/reyamir` (instead of adding it to the list of official examples `/examples/*`).

**Comment by reyamir** [comment]
I've tested latest update, everything work fine now, no more bug
I've tried all libraries, only vite + vite-plugin-ssr work best with tauri, I'll make an example later
